### PR TITLE
[MONET-1473] feat: adds hosted delivery function example for PotterDB

### PR DESCRIPTION
**Pull request by thomasspiesser**
## Purpose

Ticket https://contentful.atlassian.net/browse/MONET-1473

This PR adds an example that includes a hosted delivery function. The functions forwards GraphQL queries to the [PotterDB](https://docs.potterdb.com/apis/graphql) so that we can query Contentfuls GraphQL API for data from the potter universe.

image

### CODE REVIEW

File: examples/hosted-delivery-function-potterdb/contentful-app-manifest.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "deliveryFunctions": [
+    {
+      "id": "PotterDB",
+      "name": "Example Delivery Function querying Potter-DB GraphQL API",
```
Comment (kdamball) [inline examples/hosted-delivery-function-potterdb/contentful-app-manifest.json:5]: nit: You could move this to description and just call it "PotterDB Delivery Function"

File: examples/hosted-delivery-function-potterdb/delivery-functions/index.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  })
+
+  return {
+    namespace: 'PotterDB',
+    fields,
+  }
+}
+
+const queryHandler: EventHandler<EventType.GRAPHQL_QUERY> = async (event, context) => {
+  const response = await fetch('https://api.potterdb.com/graphql', {
```
Comment (kdamball) [inline examples/hosted-delivery-function-potterdb/delivery-functions/index.ts:21]: I would add a comment here about `isIntrospectionQuery` and how you're forwarding both to the api. Introspection query is the first thing that breaks usually 😅

**Review by kdamball — APPROVED** [review]
Only minor comments to make life easier for someone new to this

File: examples/hosted-delivery-function-potterdb/delivery-functions/index.ts (hunk 2 above)
Comment (thomasspiesser) [reply examples/hosted-delivery-function-potterdb/delivery-functions/index.ts:27]: Added a comment. Lmk if it addresses what you had in mind.

**Review by kdamball — APPROVED** [review]

**Pull request merged**